src/main/java/org/opensearch/knn/plugin/transport/KNNProfileRequest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin.transport;
+
+import lombok.Getter;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.action.support.broadcast.BroadcastRequest;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+@Getter
+public class KNNProfileRequest extends BroadcastRequest<KNNProfileRequest> {
+
+    private String index;
```

Comment:
Broadcast request should already have the indices: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/support/broadcast/BroadcastRequest.java#L52.